servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -0,0 +1,432 @@
+import tachyon.worker.block.io.LocalFileBlockWriter;
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.TempBlockMeta;
+
+/**
+ * This class represents an object store that manages all the blocks in the local tiered storage.
+ * This store exposes simple public APIs to operate blocks. Inside this store, it creates an
+ * Allocator to decide where to put a new block, an Evictor to decide where to evict a stale block,
+ * a BlockMetadataManager to maintain the status of the tiered storage, and a LockManager to
+ * coordinate read/write on the same block.
+ * <p>
+ * This class is thread-safe.
+ */
+public class TieredBlockStore implements BlockStore {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private final TachyonConf mTachyonConf;
+  private final BlockMetadataManager mMetaManager;
+  private final BlockLockManager mLockManager;
+  private final Allocator mAllocator;
+  private final Evictor mEvictor;
+  private List<BlockStoreEventListener> mBlockStoreEventListeners =
+      new ArrayList<BlockStoreEventListener>();
+
+  /** A readwrite lock for meta data **/
+  private final ReentrantReadWriteLock mEvictionLock = new ReentrantReadWriteLock();
+
+  public TieredBlockStore(TachyonConf tachyonConf) throws IOException {
+    mTachyonConf = Preconditions.checkNotNull(tachyonConf);
+    mMetaManager = BlockMetadataManager.newBlockMetadataManager(mTachyonConf);
+    mLockManager = new BlockLockManager(mMetaManager);
+
+    AllocatorType allocatorType =
+        mTachyonConf.getEnum(Constants.WORKER_ALLOCATE_STRATEGY_TYPE, AllocatorType.DEFAULT);
+    mAllocator = AllocatorFactory.create(allocatorType, mMetaManager);
+
+    EvictorType evictorType =
+        mTachyonConf.getEnum(Constants.WORKER_EVICT_STRATEGY_TYPE, EvictorType.DEFAULT);
+    mEvictor = EvictorFactory.create(evictorType, mMetaManager);
+    if (mEvictor instanceof BlockStoreEventListener) {
+      registerBlockStoreEventListener((BlockStoreEventListener) mEvictor);
+    }
+  }
+
+  @Override
+  public long lockBlock(long userId, long blockId) throws IOException {
+    return mLockManager.lockBlock(userId, blockId, BlockLockType.READ);
+  }
+
+  @Override
+  public void unlockBlock(long lockId) throws IOException {
+    mLockManager.unlockBlock(lockId);
+  }
+
+  @Override
+  public void unlockBlock(long userId, long blockId) throws IOException {
+    mLockManager.unlockBlock(userId, blockId);
+  }
+
+  @Override
+  public BlockWriter getBlockWriter(long userId, long blockId) throws IOException {
+    TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+    return new LocalFileBlockWriter(tempBlockMeta);
+  }
+
+  @Override
+  public BlockReader getBlockReader(long userId, long blockId, long lockId) throws IOException {
+    mLockManager.validateLockId(userId, blockId, lockId);
+    BlockMeta blockMeta = mMetaManager.getBlockMeta(blockId);
+    return new LocalFileBlockReader(blockMeta);
```

Comment:
Remote read are supported in DataServer, which will get the block reader and return data read.